### PR TITLE
RWA 977 - Perform Message Readiness checks

### PR DESCRIPTION
**Pull request by sullivangareth**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RWA-977

### Change description ###

Perform a readiness check to move messages from the "new" state to the "ready" state

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by dharmendrak — DISMISSED** [review]

**Review by rafalkalita — DISMISSED** [review]

**Review by rasikasri — DISMISSED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        return mockMvc.perform(put("/messages/" + messageId + "?from_dlq=false")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(getCaseEventMessage()))
+                .andExpect(status().isCreated())
+                .andReturn();
+    }
+
+    private static Stream<Arguments> provideDlqStateAndExpectedMessageState() {
+        return Stream.of(
+                Arguments.of(DLQ_EMPTY, MessageState.READY),
```
Comment (BenNzewi) [inline src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessTest.java:89]: Just trying to clarify;
Should this be DLQ_EMPTY, MessageState.NEW?

As per AC in the ticket?

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/MessageReadinessConsumer.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        // TODO Alternatively could get a list of message Ids of messages containing userId and pass
+        // that to bulk update method
+        allMessageInNewState.stream()
+                .filter(msg -> launchDarklyFeatureFlagProvider
+                        .getBooleanValue(FeatureFlag.DLQ_DB_PROCESS, getUserId(msg)))
+                .forEach(this::checkMessageToMoveToReadyState);
+    }
+
+    private void checkMessageToMoveToReadyState(CaseEventMessageEntity messageInNewState) {
+        if (deadLetterQueuePeekService.isDeadLetterQueueEmpty()) {
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/MessageReadinessConsumer.java:48]: can we add unit test to cover when this is not empty?

File: src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessTest.java (hunk 1 above)
Comment (BenNzewi) [reply src/integrationTest/java/uk/gov/hmcts/reform/wacaseeventhandler/MessageReadinessTest.java:89]: It's the expected state. This is resolved. 👍

File: src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/MessageReadinessConsumer.java (hunk 2 above)
Comment (sullivangareth) [reply src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/MessageReadinessConsumer.java:48]: Fixed in https://github.com/hmcts/wa-case-event-handler/pull/218/commits/24188312c85cd5b24bba2b8c69e1d909286b7e97
